crates/router/src/lib.rs

```diff
@@ -98,22 +115,36 @@ fn apply_request_transformations(
     Ok(bindings::new_request(new, Some(request.consume().unwrap())))
 }
 
-const FULL_URL: &[&str] = &["SPIN-FULL-URL", "X-FULL-URL"];
-const PATH_INFO: &[&str] = &["SPIN-PATH-INFO", "PATH-INFO"];
-const MATCHED_ROUTE: &[&str] = &["SPIN-MATCHED-ROUTE", "X-MATCHED-ROUTE"];
-const COMPONENT_ROUTE: &[&str] = &["SPIN-COMPONENT-ROUTE", "X-COMPONENT-ROUTE"];
-const RAW_COMPONENT_ROUTE: &[&str] = &["SPIN-RAW-COMPONENT-ROUTE", "X-RAW-COMPONENT-ROUTE"];
-const BASE_PATH: &[&str] = &["SPIN-BASE-PATH", "X-BASE-PATH"];
-const CLIENT_ADDR: &[&str] = &["SPIN-CLIENT-ADDR", "X-CLIENT-ADDR"];
-/// Calculate the default headers for the given request.
-fn calculate_default_headers<'a>(
+const FULL_URL: [&str; 2] = ["SPIN_FULL_URL", "X_FULL_URL"];
+const PATH_INFO: [&str; 2] = ["SPIN_PATH_INFO", "PATH_INFO"];
+const MATCHED_ROUTE: [&str; 2] = ["SPIN_MATCHED_ROUTE", "X_MATCHED_ROUTE"];
+const COMPONENT_ROUTE: [&str; 2] = ["SPIN_COMPONENT_ROUTE", "X_COMPONENT_ROUTE"];
+const RAW_COMPONENT_ROUTE: [&str; 2] = ["SPIN_RAW_COMPONENT_ROUTE", "X_RAW_COMPONENT_ROUTE"];
+const BASE_PATH: [&str; 2] = ["SPIN_BASE_PATH", "X_BASE_PATH"];
+const CLIENT_ADDR: [&str; 2] = ["SPIN_CLIENT_ADDR", "X_CLIENT_ADDR"];
+
+pub(crate) fn calculate_default_headers(
     req: &IncomingRequest,
-    raw_route: String,
-) -> anyhow::Result<Vec<(&'a [&'a str], String)>> {
+    base: &str,
```

Comment:
nit: I wonder if `base_route` would be clearer. What do you think?

Reply:
`base` is used pretty systematically in Spin.  I'd be in favour of renaming a whole bunch of stuff to be less terse but I know others feel differently.